src/ProjectSystemTools/BuildLogging/Model/Frontend/FrontendBuildTableDataSource.cs

```diff
@@ -143,9 +133,19 @@ public ITableEntriesSnapshot GetSnapshot(int versionNumber)
             return null;
         }
 
-        public void AddEntry(Build build)
+        public string RetrieveLogForBuild(int buildID)
         {
-            _entries = _entries.Add(build);
+            return _loggerService.RetrieveLogForBuild(buildID);
+        }
+
+        private void UpdateEntries()
+        {
+            ImmutableList<IBuildSummary> newData = _loggerService.RetrieveAllBuilds();
+            _entries = ImmutableList<UIBuildSummary>.Empty;
+            foreach (IBuildSummary summary in newData)
+            {
+                _entries = _entries.Add(new UIBuildSummary(summary));
+            }
```

Comment:
I think this can be written as something like:

```suggestion
            _entries = _loggerService
                .RetrieveAllBuilds()
                .Select(summary => new UIBuildSummary(summary))
                .ToImmutableList();
```